src/lib/storage/table.hpp

```diff
@@ -95,11 +95,13 @@ class Table : private Noncopyable {
   /**
    * Creates a new Chunk and appends it to this table.
    * Makes sure the @param segments match with the TableType (only ReferenceSegments or only data containing segments)
-   * En/Disables MVCC for the Chunk depending on whether MVCC is enabled for the table (has_mvcc())
    * This is a convenience method to enable automatically creating a chunk with correct settings given a set of segments.
-   * @param alloc
+   *
+   * @param mvcc_data   If the Table uses MVCC, this can optionally be passed in. If the Table uses MVCC and no
+   *                    mvcc_data is supplied, MvccData will be allocated.
```

Comment:
> If the Table uses MVCC and no mvcc_data is supplied, MvccData will be allocated.

No, it will not in Debug, it will fail. See `table.cpp`:
````
DebugAssert(_type != TableType::Data || static_cast<bool>(mvcc_data) == (_use_mvcc == UseMvcc::Yes)
````
I would prefer to get rid of the `DebugAssert`.

Reply:
It's actually the comment that was wrong. Updated to `@param mvcc_data   Has to be passed in iff the Table is a data Table which uses MVCC`